lib/dialects/abstract/query-generator.js

```diff
@@ -1386,16 +1386,34 @@ const QueryGenerator = {
         }
         mainQueryItems.push(this.selectFromTableFragment(options, model, mainAttributes, '('+
           options.groupedLimit.values.map(value => {
-            const where = _.assign({}, options.where);
-            where[options.groupedLimit.on] = value;
+            let where = options.where || {}
+              , include;
+
+            if (Array.isArray(options.groupedLimit.on)) {
+              include = Model._validateIncludedElements({
+                include: [{
+                  attributes: [],
+                  association: options.groupedLimit.on[0],
+                  where: {
+                    [options.groupedLimit.on[1]]: value
+                  }
+                }],
+                model
+              }).include;
+            } else {
+              where = _.assign(options.where);
+              where[options.groupedLimit.on] = value;
+            }
```

Comment:
`on` is used only for defining a where statement.
Rather than having `on` as an array could we not just support having it null and then defining an arbitrary include which could have required + where?